File: e2e/cypress-javascript/providers/SupportProvider.js
const { ApiClient } = require('@/support/api/ApiClient');
const { WireMockClient } = require('@/support/mocks/WireMockClient');
const { ApiRouteMock } = require('@/support/mocks/ApiRouteMock');

class SupportProvider {
  constructor() {
    this.api = new ApiClient();
    this.wiremock = new WireMockClient();
    this.mock = { api: new ApiRouteMock() };
  }
}

module.exports = { SupportProvider };

File: e2e/capybara-ruby/providers/support_provider.rb
# frozen_string_literal: true

require_relative '../support/api/api_client'
require_relative '../support/mocks/api_route_mock'
require_relative '../support/mocks/wire_mock_client'

class SupportProvider
  attr_reader :api, :wiremock, :mock

  def initialize
    @api = ApiClient.new
    @wiremock = WireMockClient.new
    @mock = MockFacade.new
  end

  class MockFacade
    attr_reader :api

    def initialize
      @api = ApiRouteMock.new
    end
  end
end
